addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/PolicyDefinitionUnitTest.java

```diff
@@ -349,4 +355,26 @@ void shouldReturnNullPolicyIfDefinitionYamlIsEmptyOrNullObject(String defnYamlSt
         assertThat(progress.hasWarnings(), is(equalTo(false)));
         assertThat(policyDefinition.getScanPolicy("test", progress), is(nullValue()));
     }
+
+    @ParameterizedTest
+    @ValueSource(ints = {0, 1, 50000, 99999})
+    void shouldAlwaysIncludeIDWhenSerializingARule(int id) throws JsonProcessingException {
+        // Given
+        Rule rule =
+                new Rule(
+                        id,
+                        "Test Rule",
+                        AttackStrength.MEDIUM.name(),
+                        AlertThreshold.MEDIUM.name());
+        FilterProvider filters =
+                new SimpleFilterProvider()
+                        .addFilter(DefaultPropertyFilter.FILTER_ID, new DefaultPropertyFilter());
+        String ruleYaml =
+                AutomationPlan.YAML_OBJECT_MAPPER.writer(filters).writeValueAsString(rule);
+        Yaml yaml = new Yaml();
+        Object data = yaml.load(ruleYaml);
+        LinkedHashMap<?, ?> ruleMap = (LinkedHashMap<?, ?>) data;
+        // When / Then
+        assertThat((Integer) ruleMap.get("id"), is(notNullValue()));
```

Comment:
A contains on the serialized form is more clear on what the test is verifying and better if it actually checks the values serialized.